src/io/flutter/run/FlutterAppState.java

```diff
@@ -79,16 +83,15 @@ protected ProcessHandler startProcess() throws ExecutionException {
     ConnectedDevice device = null;
 
     // Only pass the current device in if we are showing the device selector.
-    final FlutterSettings settings = FlutterSettings.getInstance(project);
-    if (settings.isShowDevices()) {
+    if (service.isActive()) {
       final Collection<ConnectedDevice> devices = service.getConnectedDevices();
       if (devices.isEmpty()) {
         throw new ExecutionException("No connected device");
       }
       device = service.getSelectedDevice();
     }
 
-    final FlutterRunnerParameters parameters = ((FlutterRunConfiguration)getEnvironment().getRunProfile()).getRunnerParameters().clone();
+    final FlutterRunnerParameters parameters = ((FlutterRunConfigurationBase)getEnvironment().getRunProfile()).getRunnerParameters().clone();
```

Comment:
Actually, looking at `FlutterAppStateBase` this looks safe.  Does it make sense to override `getEnvironment()` and add a `@NotNull` annotation to make this obvious?